test/RibbonVolatility.js

```diff
@@ -514,14 +711,14 @@ function behavesLikeRibbonVolatility(params) {
           this.buyInstrumentParams,
           {
             from: user,
-            value: this.totalPremium,
+            value: this.premiumBuffered,
             gasPrice: this.gasPrice,
           }
         );
         const receipt = await res.wait();
         console.log(receipt.gasUsed.toString());
 
-        assert.isAtMost(receipt.gasUsed, 1400000);
+        assert.isAtMost(receipt.gasUsed, 3000000);
```

Comment:
Originally at most 1400000.

Charm ETH straddle -> 1901648
Charm WBTC straddle -> 2690951

So had to bump it up to 3000000. Please lmk if you see any optimizations in purchase() for charm adapter.

Reply:
To make testing this more accurate, we need to measure the cost after doing `populateMarkets` beforehand. It should be <2000000.

Reply:
in ribbonvolatility we do not do a populateMarkets offchain. we only do it onchain if we cant find the address of the token with the given option terms. however had already called populateMarkets before the block number, so purchase should never get into [this](https://github.com/ribbon-finance/structured-products/pull/86/files#diff-c725dc65d7fdf5935bf162c13830b03a9045b83f61a4e3cb93e33680958dd7a9R186) if statement.